### PR TITLE
Data and networking

### PR DESCRIPTION
**Pull request by BenGoldstein88**
First pass stuff as well as beginning of networking

Create-React-App to scaffold things.
Theme hooks for heritable styling through Material UI

/pages for any page with its own route
/components for anything (potentially) reusable
/contexts for shared application state that any number of components may need to read/write

Test networking in UsersAndTeams.js. As of now, uploading a CSV creates teams (but not users) based on contents. Hooks to TeamsContext and correctly renders changes on change.

### CODE REVIEW

File: src/pages/UsersAndTeams.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    reader.readAsText(file);
+  };
+
+  const parseFileInput = file => {
+    console.log("FILE", file);
+  };
+
+  const createTeams = teams => {
+    console.log("CREATING TEAMS", teams);
+    const url = `https://api.handstandwith.us/v2/events/${eventSlug}/teams`;
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:63]: this base url should pull from the current ENV. Maybe a url constructor that takes in the path (e.g. `teams`)

File: src/contexts/UsersContext.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+export const initialUsersState = {
+  users: [],
+  selectedUser: {}
+};
+
+const EDIT = "EDIT";
+const REPLACE = "REPLACE";
```
Comment (samgold89) [inline src/contexts/UsersContext.js:8]: Not exactly sure which functionality these `REPLACE`s represent, but also recognize they're placeholder :)

File: src/pages/UsersAndTeams.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FileDropzone } from "../components/FileDropzone";
+
+import camelCase from "lodash/camelCase";
+import { TeamsContext, editTeams } from "../contexts/TeamsContext";
+export const UsersAndTeams = props => {
+  const proxyUrl = "https://cors-anywhere.herokuapp.com/";
+  // const proxyUrl = "";
+  const headers = {
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:10]: Let's not check these in, or somehow obfuscate them. Yaml file or something?

Comment (samgold89) [reply src/pages/UsersAndTeams.js:10]: or I guess they'll eventually be pulled in from the authentication, but let's keep 'em out of this for now.

File: src/pages/UsersAndTeams.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          //   console.log("RESPONSE", response);
+          return response.json();
+        })
+        .then(json => {
+          //   console.log("HELLO FROM TEAMS", json);
+        });
+    });
+  };
+
+  const updateFromCsv = users => {
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:87]: Would something like this make this cleaner, more one-liner-y?
`Array.from(new Set(ALL-TEAM-NAMES));`

File: src/pages/UsersAndTeams.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        }
+      })
+      .map(user => {
+        return user.teamName;
+      });
+
+    console.log("U", uniqueTeams);
+
+    createTeams(uniqueTeams);
+    const url = `https://api.handstandwith.us/v2/events/${eventSlug}/teams`;
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:103]: ditto to above

File: src/theme.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    green: {
+      main: "#1DC57C",
+      light: "#DBF5EA"
+    },
+    red: {
+      main: "#F20000",
+      light: "#F5DBDB"
+    },
+    blue: {
+      main: "#002BFF",
```
Comment (samgold89) [inline src/theme.js:58]: #00d5ff

File: src/theme.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      main: "#dd7727"
+    },
+    grey: {
+      main: "#F7F6F5",
+      dark: "#939393",
+      border: "#ECE5DF",
+      placeholder: "#A6A6A6"
+    },
+    green: {
+      main: "#1DC57C",
```
Comment (samgold89) [inline src/theme.js:50]: #3ecf8e

File: src/theme.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      light: "#DBF5EA"
+    },
+    red: {
+      main: "#F20000",
```
Comment (samgold89) [inline src/theme.js:54]: #ff4200

File: src/contexts/UsersContext.js (hunk 2 above)
Comment (BenGoldstein88) [reply src/contexts/UsersContext.js:8]: The way it's written, "EDIT" lets you overwrite a single key in the context while "REPLACE" lets you replace or clear the entire context.

I will probably just rewrite to have edit merge the old and new keys. Just have to see where the app data takes us.

File: src/pages/UsersAndTeams.js (hunk 1 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:63]: 👍

File: src/pages/UsersAndTeams.js (hunk 3 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:10]: 👍

File: src/pages/UsersAndTeams.js (hunk 4 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:87]: 👍

File: src/pages/UsersAndTeams.js (hunk 5 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:103]: 👍

File: src/theme.js (hunk 6 above)
Comment (BenGoldstein88) [reply src/theme.js:58]: 👍

File: src/theme.js (hunk 7 above)
Comment (BenGoldstein88) [reply src/theme.js:50]: 👍

File: src/theme.js (hunk 8 above)
Comment (BenGoldstein88) [reply src/theme.js:54]: 👍

**Pull request merged**